uefi/flash.go

```diff
@@ -139,6 +201,85 @@ func (f *FlashImage) Extract(dirPath string) error {
 	return ioutil.WriteFile(jsonPath, b, 0666)
 }
 
+// Assemble assembles the FlashImage starting from the bottom up.
+func (f *FlashImage) Assemble() ([]byte, error) {
+	// Assemble the ifd
+	ifdbuf, err := f.IFD.Assemble()
+	if err != nil {
+		return nil, err
+	}
+	// Assemble regions.
+	// We need to sort them since a) we don't really know the order until we parse the block numbers
+	// and b) the order may have changed anyway.
+	if !f.IFD.Region.BIOS.Valid() {
+		return nil, fmt.Errorf("no BIOS region: invalid region parameters %v", f.IFD.Region.BIOS)
+	}
+	type region struct {
+		P   *Region
+		buf []byte
+	}
+	regions := make([]region, 0, 0)
```

Comment:
I can see the length as 0, but why not make the capacity something reasonable, or just not set it? Or just do
var regions []region
if you're going to make it zero size?
If you do this:
regions := make([]region,0, 32) 
then the append will be pretty efficient until you hit 32. No reallocations until you hit it.

Reply:
ok will do

Reply:
There are max 4 regions. I'll make cap 4.